### ISSUE TITLE
Dock shows up when drop down terminal is triggered

### ISSUE THREAD

**Issue by navdeepio** [micheleg/dash-to-dock#592]
With intellihide enabled!

**Comment by micheleg** [comment]
Hi,

i need some more specific information: which drop down terminal are we talking about? Which gnome-shell and extension version?

Thaks

**Comment by navdeepio** [comment]
Hi, sorry about the brief description.

I'm using the latest gnome-shell with drop down terminal gnome extension(latest version)

**Comment by micheleg** [comment]
I'm wondering if this is some sort of regression: did this use to work in the past?

My understading is that:
1. With intelligent autohide enabled, the dock correctly hide when the terminal is shown, as we explicitely support this https://github.com/micheleg/dash-to-dock/blob/c6c56866e6120c30a36a23f2ff630a2f90f925b0/intellihide.js#L265-L268
2. With intelligent autohide disabled, i.e. with fixed dock, the terminal is indeed behind the dock, in the same way that it is behind the top panel. The terminal is effectively another window, and belongs to a different layer in the sheel UI. I don't think I can solve the issue on my side. I would recommend you to submit a bug to the dropdownterminal extension. My suggestion would be that they track the workArea when placing the terminal window, so that it doesn't lays behind the dock anymore.

**Comment by navdeepio** [comment]
Not sure about the past behavior but here are my observations:

1. When the terminal is triggered with the dock already hidden(intellihide), the dock shows up!

2. In a workspace with the dock visible, triggering the terminal doesn't hide the dock.

Please let me know if you need my dash to dock settings etc

**Comment by micheleg** [comment]
I cannot reproduce the issue: can you confirm that you are using this extension https://extensions.gnome.org/extension/442/drop-down-terminal/ and not another similar one?

**Comment by navdeepio** [comment]
@micheleg Yes, I'm using the said extension.

If you want I could attach a screencast..

**Comment by micheleg** [comment]
Any additional information would be useful: settings (```dconf dump /org/gnome/shell/extensions/dash-to-dock/```), exact extension and gnome shell version, wayland/X11, monitor configuration (i.e. multi monitor). 

Please, also check disabling all other extensions other than dash to dock and drop down terminal.

**Comment by roachsinai** [comment]
Same issue. And my dropdown terminal is Guake.

+ OS: Manjaro 17.1.8
+ Gnome version: GNOME Shell 3.28.1
+ Guake version: Guake Terminal 3.0.5

And I have set a Manual keybinding:

> If you want to trigger guake manually, for instance on system where libkeybinder3 does not work, you > can register the following snippet in your window manager

```
dbus-send --type=method_call --dest=org.guake3.RemoteControl \
    /org/guake3/RemoteControl org.guake3.RemoteControl.show_hide
```

You can use the simpler

`guake -t`

![image](https://user-images.githubusercontent.com/9500049/39011601-7e78f2bc-4444-11e8-8b2b-b27defc2e9ee.png)

**Comment by roachsinai** [comment]
Here is my `dconf dump /org/gnome/shell/extensions/dash-to-dock/`.

```
[/]
show-apps-at-top=false
shift-click-action='minimize'
shift-middle-click-action='launch'
extend-height=false
dock-fixed=false
intellihide-mode='FOCUS_APPLICATION_WINDOWS'
multi-monitor=false
custom-background-color=true
apply-custom-theme=false
dock-position='BOTTOM'
height-fraction=1.0
preferred-monitor=0
running-indicator-style='DOTS'
dash-max-icon-size=39
background-color='#eeeeec'
middle-click-action='launch'
transparency-mode='ADAPTIVE'
```

Could you please give me some advices? @micheleg

**Comment by micheleg** [comment]
@roachsinai Could you check if the setting the intellihide mode to "all windows" help?

**Comment by ocroquette** [comment]
I just had the same issue, and setting the intellihide mode to "all windows" fixed it.
If I understand correctly, it is just a coincidence that it happened to me with the Terminal just like to [navxio](https://github.com/navxio). So this is not a bug, "autohide" works as designed. The question is rather if "all windows" should be the default setting. It is indeed surprising to see the dock appear when switching to another application with ALT+Tab.

**Comment by sergio-costas** [comment]
Quick question: was the terminal window covering the dock? I mean: by default, the dock hides only if the _active_ window covers it.

**Comment by ocroquette** [comment]
No, it happened to me when switching from a window that covered the dock (a text editor) to the terminal that did not cover the dock. Then the dock surprisingly appeared.
